packages/integration-tests/src/EntityManager.test.ts

```diff
@@ -1023,6 +1023,95 @@ describe("EntityManager", () => {
     // Then the deleted entity was returned from the flush
     expect(result).toEqual(a1);
   });
+
+  it("can clone entities", async() => {
+    const em = newEntityManager();
+
+    // Given an entity
+    const a1 = new Author(em, {firstName: "a1"});
+    await em.flush();
+
+    // When we clone that entity
+    const a2 =  await em.cloneEntity(a1);
+
+    // Then we expect the cloned entity to have the same properties as the original
+    expect(a2.firstName).toEqual(a1.firstName);
+    expect(a2.id).not.toEqual(a1.id)
+  });
+
+  it("can clone entities and referenced entities", async () => {
+    const em = newEntityManager();
+
+    // Given an entity with a reference to another entity
+    const a1 = new Author(em, {firstName: "a1"});
+    const b1 = new Book(em, {title: "b1", author: a1});
+    await em.flush();
+
+    // When we clone that entity and its reference
+    const a2 =  await em.cloneEntity(a1, "books");
+    await em.flush();
+
+    // Then we expect the cloned entity to have a cloned copy of the original's reference
+    expect(a2.books.get[0].title).toEqual(b1.title);
+    expect(a2.books.get[0].id).not.toEqual(b1.id);
+  });
+
+  it("cannot clone many-to-many references", async () => {
+    const em = newEntityManager();
+
+    // Given an entity with a reference to another entity with a many-to-many reference
+    const a1 = new Author(em, {firstName: "a1"});
+    const b1 = new Book(em, {title: "b1", author: a1});
+    const t1 = new Tag(em, {name: "t1", books: [b1]})
+    await em.flush();
+
+    // When we clone that entity and its nested references
+    const toExecute = async function () { await em.cloneEntity(a1, {books: "tags"})};
+
+    // Then we expect the cloning to fail
+    await expect(toExecute).rejects.toThrow()
+  });
+
+  it("can clone nested references", async () => {
+    const em = newEntityManager();
+
+    // Given an entity with a reference to another entity which references a third entity
+    const a1 = new Author(em, {firstName: "a1"});
+    const b1 = new Book(em, {title: "b1", author: a1});
+    const i1 = new Image(em, {fileName: "11", type: ImageType.BookImage, book: b1});
+    await em.flush();
+
+    // When we clone that entity and its nested references
+    const toExecute = async function () { await em.cloneEntity(a1, {books: "tags"})};
+
+    // Then we expect the cloning to fail
+    await expect(toExecute).rejects.toThrow()
+    const a2 =  await em.cloneEntity(a1, {books: "image"});
+    await em.flush();
+
+    // Then we expect the cloned entity to have cloned copies of all its nested references
+    const b2 = (await a2.books.load())[0];
+    const i2 = await b2.image.load();
+    expect(i2).toBeTruthy();
+    expect(i2?.id).not.toEqual(i1.id);
+    expect(i2?.fileName).toEqual(i1.fileName);
+    expect(i2?.type).toEqual(i1.type);
```

Comment:
Can we add assertions about the overall number of entities in the system? I.e. assert `find(Author, {})` == 2, number of books = 2, number of images = 2 (if that is right I'm kinda guessing).

Just to get a 2nd triangulation for "we created the right number of new entities".

...I'd be tempted to include this "number of authors, books, reviews is now X" so it might be worth a helper method, but defer. 

like `expect(await numberOfAuthors(em)).toEqual(2)` or something.

Maybe `expect(await numberOf(Author, Book, Review)).toEqual([2, 3, 4])` if we wanted to be really cute.

Reply:
I gave it a shot! I just declared the helper at the bottom of this file.